packages/grid/_modules_/grid/hooks/features/core/useGridState.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { ApiRef } from '../../../models/api/apiRef';
+import { useRafUpdate } from '../../utils/useRafUpdate';
+import { GridState } from './gridState';
+import { useGridApi } from './useGridApi';
+
+export const useGridState = (
+  apiRef: ApiRef,
+): [GridState, (stateUpdaterFn: (oldState: GridState) => GridState) => void, () => void] => {
+  const api = useGridApi(apiRef);
+  const [, forceUpdate] = React.useState();
+  const [rafUpdate] = useRafUpdate(() => forceUpdate((p: any) => !p));
```

Comment:
there is a reason why I use raf and do all this state layer. If I use useState then it will rerender at every state change and this adds a massive performance issue.